### PR TITLE
Feature lwdev 7722 instruments advanced search

### PR DESCRIPTION
**Pull request by GeorgePuisha**
https://lykkex.atlassian.net/browse/LWDEV-7722

### CODE REVIEW

File: src/stores/referenceStore.ts [hunk 1]
```diff
@@ -115,32 +123,60 @@ class ReferenceStore extends BaseStore {
     }
   };
 
+  findAppropriateDescriptionById = (
+    descriptions: any[],
+    assetId: string
+  ): DescriptionResponseModel => {
+    return (
+      descriptions.find(
+        (rawDescription: any) => rawDescription.Id === assetId
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: ```any -> string``` ?

File: src/stores/referenceStore.ts [hunk 2]
```diff
@@ -115,32 +123,60 @@ class ReferenceStore extends BaseStore {
...
-          if (!dtoAssets) {
-            return;
-          }
-          runInAction(() => {
-            this.assets = dtoAssets.map((x: any) =>
-              mappers.mapToAsset(x, this.categories)
+    const requests = [this.api.fetchAll(), this.api.fetchAssetsDescriptions()];
+
+    return Promise.all(requests).then(data => {
+      const assets = data[0].Assets || data[0];
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: Add TODO

File: src/stores/referenceStore.ts [hunk 3]
```diff
@@ -115,32 +123,60 @@ class ReferenceStore extends BaseStore {
...
+      this.descriptions = data[1].Descriptions || data[1];
+      if (assets && assets.length > 0) {
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: assets && assets.length

File: src/stores/referenceStore.ts [hunk 4]
```diff
@@ -115,32 +123,60 @@ class ReferenceStore extends BaseStore {
...
 
   fetchAssetById = (id: string) => {
-    return this.api.fetchAssetById(id).then((resp: any) => {
-      const dtoAsset = resp.Asset || resp;
-      const asset = mappers.mapToAsset(dtoAsset, this.categories);
-      this.assets.push(asset);
-      return Promise.resolve(asset);
+    return this.api.fetchAssetById(id).then(asset => {
+      let mappedAsset;
+      asset = asset.Asset || asset;
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: Maybe it make sense to push result into new variable with different name just for better readability?

File: src/models/instrumentModel.ts [hunk 5]
```diff
@@ -25,6 +25,14 @@ class InstrumentModel {
     return undefined;
   }
 
+  @computed
+  get invertedDisplayName() {
+    if (this.baseAsset && this.quoteAsset) {
+      return join('/', [this.quoteAsset.name, this.baseAsset.name]);
+    }
+    return undefined;
```
Comment (PaulRumyantsevItr) [inline src/models/instrumentModel.ts]: Redundant, function will return undefined by default

File: src/stores/referenceStore.ts [hunk 6]
```diff
@@ -115,32 +123,62 @@ class ReferenceStore extends BaseStore {
...
 
   fetchAssetById = (id: string) => {
-    return this.api.fetchAssetById(id).then((resp: any) => {
-      const dtoAsset = resp.Asset || resp;
-      const asset = mappers.mapToAsset(dtoAsset, this.categories);
-      this.assets.push(asset);
-      return Promise.resolve(asset);
+    return this.api.fetchAssetById(id).then(asset => {
+      let mappedAsset;
+      const rawAsset = asset.Asset || asset;
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: Are you sure this one is raw, not the one from parameters?

File: src/models/instrumentModel.ts (hunk 5 above)
Comment (GeorgePuisha) [reply src/models/instrumentModel.ts]: Deny, all code paths should return a string or undefined according to tslint suggestion.

**Pull request merged**